examples/26_planning_agent_workflow.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+"""
+Planning Agent Workflow Example
+
+This example demonstrates a two-stage workflow:
+1. Planning Agent: Analyzes the task and creates a detailed implementation plan
+2. Execution Agent: Implements the plan with full editing capabilities
+
+The task: Create a Python web scraper that extracts article titles and URLs
+from a news website, handles rate limiting, and saves results to JSON.
+"""
+
+import tempfile
+from pathlib import Path
+
+from pydantic import SecretStr
+
+from openhands.sdk import LLM, Conversation
+from openhands.sdk.llm import content_to_str
+from openhands.tools.preset import get_planning_agent
+
+
+def get_event_content(event):
+    """Extract content from an event."""
+    if hasattr(event, "llm_message"):
+        return "".join(content_to_str(event.llm_message.content))
+    return str(event)
+
+
+def main():
+    """Run the planning agent workflow example."""
+
+    # Create a temporary workspace
+    workspace_dir = Path(tempfile.mkdtemp())
+    print(f"Working in: {workspace_dir}")
+
+    # Initialize LLM (you'll need to set your API key)
+    llm = LLM(
+        model="gpt-4o-mini",
+        api_key=SecretStr("your-api-key-here"),  # Replace with your actual API key
+        service_id="openai",
+    )
```

Comment:
please run it end to end to make sure it works and set the model name here similar to other examples